### PR TITLE
[onert/nnfw_api] Add unit tests for RoPE op

### PR DESCRIPTION
**Pull request by ys44kim**
This commit adds unit tests for RoPE

ONE-DCO-1.0-Signed-off-by: youngsik kim <ys44.kim@samsung.com>

issue : https://github.com/Samsung/ONE/issues/14108
draft : https://github.com/Samsung/ONE/pull/14090

### CODE REVIEW

File: tests/nnfw_api/lib/CircleGen.cc [hunk 1]
```diff
@@ -596,6 +596,13 @@ uint32_t CircleGen::addOperatorRmsNorm(const OperatorParams &params, float epsil
                                 circle::BuiltinOptions_RmsNormOptions, options);
 }
 
+uint32_t CircleGen::addOperatorRoPE(const OperatorParams &params)
```
Comment (glistening) [inline tests/nnfw_api/lib/CircleGen.cc:599]: @ys44kim Could you please add `mode` param?

File: tests/nnfw_api/src/GenModelTests/one_op_tests/RoPE.test.cc [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+  uint32_t sin_table_buf = cgen.addBuffer(std::vector<float>{0.5, 1.0, 1.0, 0.5});
+  int sin_table =
+    cgen.addTensor({{1, 1, 1, 4}, circle::TensorType::TensorType_FLOAT32, sin_table_buf});
+  uint32_t cos_table_buf = cgen.addBuffer(std::vector<float>{1.0, 0.5, 0.5, 1.0});
+  int cos_table =
+    cgen.addTensor({{1, 1, 1, 4}, circle::TensorType::TensorType_FLOAT32, cos_table_buf});
+  int in = cgen.addTensor({{1, 1, 1, 4}, circle::TensorType::TensorType_FLOAT32});
+  int out = cgen.addTensor({{1, 1, 1, 4}, circle::TensorType::TensorType_FLOAT32});
+
+  cgen.addOperatorRoPE({{in, sin_table, cos_table}, {out}});
```
Comment (glistening) [inline tests/nnfw_api/src/GenModelTests/one_op_tests/RoPE.test.cc:31]: and specify mode `GPT_NEOX` here.

File: tests/nnfw_api/lib/CircleGen.cc (hunk 1 above)
Comment (ys44kim) [reply tests/nnfw_api/lib/CircleGen.cc:599]: Currently, only one mode (GPT-NEOX) is supported, so I thought the default value was used. i didn't care about mode.
I argee it's good to add mode like your opinion.
i will add `mode`.

**Comment by ys44kim** [comment]
@glistening 
i updated this PR.
please review it.

**Review by zetwhite — APPROVED** [review]
LGTM

**Review by seockho-kim — APPROVED** [review]
LGTM

**Review by glistening — APPROVED** [review]

**Pull request merged**